health.go

```diff
@@ -80,8 +80,19 @@ type Health interface {
 	// DeregisterAll Deregister removes all health checks from this instance, and stops their next executions.
 	// It is equivalent of calling Deregister() for each currently registered check.
 	DeregisterAll()
-	// WithLogger allows you to change the logging implementation, defaults to standard logging
-	WithLogger(logger log.Logger)
+	// WithCheckListener allows you to listen to check start/end events
+	WithCheckListener(listener CheckListener)
+}
+
+// CheckListener can be used to gain check stats or log check transitions.
+// Implementations of this interface **must not block**
```

Comment:
I would add a bit more explanation to this. Suggestion: 

```
Implementations of this interface **must not block**
If an implementation blocks, it may result in delayed exacution of other healthchecks down the line.
It's OK to log in the implementation and it's OK to add metrics, but it's not OK to run anything that takes long time to complete such as network IO etc
```